### PR TITLE
rename FOO.BAR packages to FOO/BAR

### PR DESCRIPTION
**Pull request by stylewarning**
This is a heavy renaming of all of our `foo.bar` package names to `foo/bar`.

The reason is that in old CMUCL and in Allegro CL, the `foo.bar` notation is for hierarchical packages, a non-standard Lisp feature that gives you nested namespaces. Moreover, the discrepancy between systems using `/` and packages using `.` had no good explanation.

This also removes all `rename-package` and conditionals on package-local-nicknames. We just assume you have it now.

This cleans up some existing nicknames due to issues/conflicts in the packages.lisp file

I'm a little worried I didn't replace *everything* in various scripts, dockerfiles, etc.

See also (required): https://github.com/quil-lang/qvm/pull/294

### CODE REVIEW

File: src/package.lisp [hunk 1]
```diff
@@ -729,17 +722,23 @@
...
+  (:use #:cl)
+  (:local-nicknames (#:a      #:alexandria)
+                    (#:quilfe #:cl-quil/frontend))
+  (:import-from #:cl-quil/frontend #:tok #:token-type #:token-payload)
+
+  (:export
+   #:parse-qasm))
 
 (defpackage #:cl-quil
-  (:nicknames #:quil)
```
Comment (appleby) [inline src/package.lisp:736]: sobrave

File: src/package.lisp [hunk 2]
```diff
@@ -1353,7 +1336,7 @@
 ;;; access "internal" symbols.  Having such symbols cataloged here is
 ;;; preferable to simply accessing any symbol whatsoever via full
 ;;; qualification, ala the use of package name "quil" with double
```
Comment (appleby) [inline src/package.lisp:1338]:
```suggestion
;;; qualification, ala the use of package name "cl-quil" with double
```
boom. gottem.

File: src/chip-library/package.lisp [hunk 3]
```diff
@@ -2,7 +2,7 @@
 ;;;;
 ;;;; Author: A.J. Nyquist
 
-(defpackage #:cl-quil.chip-library
+(defpackage #:cl-quil/chip-library
   (:use #:cl)
   (:local-nicknames (:q :cl-quil)
```
Comment (appleby) [inline src/chip-library/package.lisp:7]: `:q` -> `:quil` imho

**Review by appleby — APPROVED** [review]

**Review by karlosz — DISMISSED** [review]

File: src/package.lisp (hunk 2 above)
Comment (notmgsk) [reply src/package.lisp:1338]: also `à la`

smh

**Review by notmgsk — DISMISSED** [review]
Agree w the great Appleby

File: src/package.lisp (hunk 2 above)
Comment (appleby) [reply src/package.lisp:1338]: M-x ala-mode

**Comment by stylewarning** [comment]
I'm not convinced this PR is a good idea anymore.

1. It does a bunch of different changes all in one PR.
2. It breaks `~/.../` format strings, which can't be fixed without hacks
3. It's not clear hierarchical packages are actually causing problems per se.

**Pull request closed without merging**